frameworks/Rust/actix-4/src/main_mongodb.rs

```diff
@@ -0,0 +1,215 @@
+}
+
+async fn find_random_world(data: web::Data<Data>) -> Result<World> {
+    let runtime = data.tokio_runtime.clone();
+    runtime
+        .spawn(async move {
+            let mut rng = SmallRng::from_entropy();
+            let id = (rng.gen::<u32>() % 10_000 + 1) as i32;
+
+            let coll = data.client.database("hello_world").collection("world");
+            let world = coll
+                .find_one(doc! { "id": id as f32 }, None)
+                .await?
+                .expect("should find world");
+            Ok(world)
+        })
+        .await?
+}
+
+#[actix_web::get("/db")]
+async fn db(data: web::Data<Data>) -> Result<HttpResponse<Vec<u8>>> {
+    let world = find_random_world(data).await?;
+    let mut bytes = Vec::with_capacity(48);
+    serde_json::to_writer(&mut bytes, &world)?;
+
+    let mut res = HttpResponse::with_body(StatusCode::OK, bytes);
+    res.headers_mut()
+        .insert(SERVER, HeaderValue::from_static("Actix"));
+    res.headers_mut()
+        .insert(CONTENT_TYPE, HeaderValue::from_static("application/json"));
+
+    Ok(res)
+}
+
+async fn find_random_worlds(data: web::Data<Data>, num_of_worlds: usize) -> Result<Vec<World>> {
+    let mut futs = FuturesUnordered::new();
+    for _ in 0..num_of_worlds {
+        futs.push(find_random_world(data.clone()))
+    }
+
+    let mut worlds = Vec::with_capacity(num_of_worlds);
+    while let Some(world) = futs.try_next().await? {
+        worlds.push(world);
+    }
+
+    Ok(worlds)
+}
+
+#[actix_web::get("/queries")]
+async fn queries(
+    data: web::Data<Data>,
+    query: web::Query<Queries>,
+) -> Result<HttpResponse<Vec<u8>>> {
+    let n_queries = query.q;
+
+    let worlds = find_random_worlds(data, n_queries).await?;
+
+    let mut bytes = Vec::with_capacity(35 * n_queries);
+    serde_json::to_writer(&mut bytes, &worlds)?;
+
+    let mut res = HttpResponse::with_body(StatusCode::OK, bytes);
+    res.headers_mut()
+        .insert(SERVER, HeaderValue::from_static("Actix"));
+    res.headers_mut()
+        .insert(CONTENT_TYPE, HeaderValue::from_static("application/json"));
+
+    Ok(res)
+}
+
+#[actix_web::get("/updates")]
+async fn updates(
+    data: web::Data<Data>,
+    query: web::Query<Queries>,
+) -> Result<HttpResponse<Vec<u8>>> {
+    let tokio_runtime = data.tokio_runtime.clone();
+    let client = data.client.clone();
+
+    let mut worlds = find_random_worlds(data, query.q).await?;
+
+    let mut rng = SmallRng::from_entropy();
+    let mut updates = Vec::new();
+    for world in worlds.iter_mut() {
+        let new_random_number = (rng.gen::<u32>() % 10_000 + 1) as i32;
+        updates.push(doc! {
+            "q": { "id": world.id }, "u": { "$set": { "randomNumber": new_random_number }}
+        });
+        world.random_number = new_random_number;
+    }
+
+    tokio_runtime
+        .spawn(async move {
+            client
+                .database("hello_world")
+                .run_command(
```

Comment:
once we have a proper bulk API, we should update this to use it.

Reply:
should we file a ticket/document this somewhere?

Reply:
I filed https://jira.mongodb.org/browse/RUST-1165 for this, no need to add a comment in the actual code though since it'll live in a repo we don't own.